### DISCUSSION TITLE
[4.x] New 'environment' condition type for fields

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#11049] [Plugin ideas] [answered]
**Idea:** 
I think it would be useful to have an environment variable toggle a field's visibility. 

**Implementation:** 
This would obviously be implemented in the condition builder in field layouts (maybe also in custom sources), and I suppose the environment variable would have to be of the boolean type.

**Possible use case:** 
Sometimes there are fields that act like settings and cannot be disabled in a production environment, ~~or for specific users~~ (without the use of a plugin). The option to hide them would declutter the interface and safeguard from any accidental changes. There could also be some use cases for such a condition when creating **custom sources,** so as to provide a way to have them visible for example only in a dev enviroment.

**Comment by brandonkelly** [comment, marked as answer]
I’m not sure this is something we would add directly in Craft… feels a bit more appropriate for a plugin to add.

**Reply by thupsi** [reply]
Agreed! Also, reading through my idea again, such a condition would not fit in the entry group of conditions, but would require a new group of 'environment conditions'.
